Doc/library/threading.rst

```diff
@@ -620,25 +625,38 @@ Reentrant locks also support the :ref:`context management protocol <with-locks>`
 
       Acquire a lock, blocking or non-blocking.
 
-      When invoked without arguments: if this thread already owns the lock, increment
-      the recursion level by one, and return immediately.  Otherwise, if another
-      thread owns the lock, block until the lock is unlocked.  Once the lock is
-      unlocked (not owned by any thread), then grab ownership, set the recursion level
-      to one, and return.  If more than one thread is blocked waiting until the lock
-      is unlocked, only one at a time will be able to grab ownership of the lock.
-      There is no return value in this case.
+      When invoked with the *blocking* argument set to ``True`` (the default):
+
+         1. If another thread owns the lock, block until we are able to acquire
+            lock, or *timeout*, if set.
+
+         2. If no thread owns the lock, acquire the lock and return immediately.
+
+         3. If the same thread owns the lock, acquire the lock again, and
+            return immediately. This is the difference between ``Lock`` and
+            ``RLock``. Internally, the "recursion level" is atomically
```

Comment:
Lets not mention the internal implementation details.  get rid of the "Internally, ..." sentence.

Reply:
Yeah, noticed that as well. Implemented in my suggestions

Reply:
Removed.

Note: we talk about other internal details in the beginning of the RLock, as well as the doc for `release`, that this PR otherwise did not touch.

Reply:
Seems like this is resolved, so I'll mark this as resolved and merge this, and we can iterate further if necessary in a followup. Thanks!